File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    
    ## num = list(permutations(numbers.split("")))
    answer = []
    for i in range(1, len(numbers)+1):
        nums = set(list(map("".join,permutations(numbers, i))))
        arr = [int(j) for j in list(nums)]
       
        for j in arr:
            isPrime = True if j>1 else False
            for k in range(2, int(j**(1/2)+1)):
                if(j==k): continue
                if(j%k==0):
                    isPrime=False
                    break
            ## print(j, isPrime)
            if(isPrime):
                answer.append(j)
        ### 소수인지 여부 따져라
        
    
    ##print(answer)
    return len(list(set(answer)))

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const n = numbers.length;
    const arr = numbers.split("").map(Number); // 문자열 → 숫자 배열
    let visited = Array.from({ length: n }).fill(false);
    const nums = new Set();

    function dfs(targetLength, lst) {
        if (lst.length === targetLength) {
            nums.add(Number(lst.join("")));
            return;
        }
        for (let i = 0; i < n; i++) {
            if (visited[i]) continue;
            visited[i] = true;
            dfs(targetLength, [...lst, arr[i]]);
            visited[i] = false;
        }
    }

    for (let i = 1; i <= n; i++) {
        dfs(i, []);
    }

    function isPrime(num) {
        if (num < 2) return false;
        for (let i = 2; i <= Math.sqrt(num); i++) {
            if (num % i === 0) return false;
        }
        return true;
    }

    let count = 0;
    for (let e of nums) {
        if (isPrime(e)) count++;
    }

    return count;
}
